f5_openstack_agent/lbaasv2/drivers/bigip/lbaas_builder.py

```diff
@@ -155,7 +182,34 @@ def _assure_pools_created(self, service):
                         self.pool_builder.create_pool(svc, bigips)
                     else:
                         self.pool_builder.update_pool(svc, bigips)
+                except HTTPError as err:
+                    if err.response.status_code != 409:
+                        pool['provisioning_status'] = plugin_const.ERROR
+                        loadbalancer['provisioning_status'] = \
+                            plugin_const.ERROR
+                        raise f5_ex.PoolCreationException(err.message)
+                except Exception as err:
+                    pool['provisioning_status'] = plugin_const.ERROR
+                    loadbalancer['provisioning_status'] = \
+                        plugin_const.ERROR
+                    raise f5_ex.PoolCreationException(str(err))
+
+    def _assure_pools_configured(self, service):
+        if "pools" not in service:
+            return
+
+        pools = service["pools"]
+        loadbalancer = service["loadbalancer"]
 
+        bigips = self.driver.get_config_bigips()
+
+        for pool in pools:
+            if pool['provisioning_status'] != plugin_const.PENDING_DELETE and \
+                    self._is_not_error(pool):
+                svc = {"loadbalancer": loadbalancer,
+                       "pool": pool}
```

Comment:
There's a problem with svc. It does not include a listener and it is later used to update the associated pool for a listener.

Reply:
Just need to add this:

 ` self.add_listener_pool(service, svc)`

Reply:
@jlongstaf, @richbrowne This was handled earlier in this same method.  I don't know that I ever mentioned this.  So this should be fixed and the branch ready for review.